R/combinations-helpers.R

```diff
@@ -97,15 +99,15 @@ calculate_excess <- function(metric,
   # no repeats
   # TODO: Check that there are no NAs
   idx <- S4Vectors::match(
-    DataFrame(measured[, series_identifiers]), 
-    DataFrame(metric[, series_identifiers])
+    DataFrame(measured[, ..series_identifiers]), 
+    DataFrame(metric[, ..series_identifiers])
   )
   
-  out <- measured[, series_identifiers]
-  excess <- metric[idx, metric_col] - measured[, measured_col]
-  excess[apply(as.matrix(out[, series_identifiers]) == 0, 1, any)] <- NA
+  out <- measured[, ..series_identifiers]
+  excess <- unlist(metric[idx, ..metric_col]) - unlist(measured[, ..measured_col])
+  excess[apply(as.matrix(out[, ..series_identifiers]) == 0, 1, any)] <- NA
   out$excess <- excess
-  as.data.frame(out)
+  out
```

Comment:
class(out) is now `data.table`, right? Or still DataFrame as in @return is described?